src/wp-includes/block-template-utils.php

```diff
@@ -1432,39 +1432,45 @@ function get_template_hierarchy( $slug, $is_custom = false, $template_prefix = '
 	$template_hierarchy[] = 'index';
 	return $template_hierarchy;
 }
+
 /**
  * Inject ignoredHookedBlocks metadata attributes into a template or template part.
  *
- * Given a `wp_template` or `wp_template_part` post object, locate all blocks that have
+ * Given an object that represents a `wp_template` or `wp_template_part` post object
+ * prepared for inserting or updating the database, locate all blocks that have
  * hooked blocks, and inject a `metadata.ignoredHookedBlocks` attribute into the anchor
  * blocks to reflect the latter.
  *
- * @param WP_Post $post A post object with post type set to `wp_template` or `wp_template_part`.
- * @return WP_Post The updated post object.
+ * @param stdClass        $post    An object representing a template or template part
+ *                                 prepared for inserting or updating the database.
+ * @param WP_REST_Request $request Request object.
+ * @return stdClass The updated object representing a template or template part.
```

Comment:
Good 👍  because it was wrong before

Reply:
Oh, but we were hooking to the [`rest_after_insert_{$this->post_type}` action](https://developer.wordpress.org/reference/hooks/rest_after_insert_this-post_type/) before, which passes a `WP_Post` as its first param (rather than a `stdClass`). Or did you mean something else? 🤔

Reply:
@ockham Before, the docs said it returned a post object but it actually didn‘t return anything.

Reply:
Ah, gotcha! I'd overlooked that part 😬